exchanges/gateio/gateio_convert.go

```diff
@@ -0,0 +1,160 @@
+package gateio
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+)
+
+type gateioMilliSecTime time.Time
+
+func (a *gateioMilliSecTime) UnmarshalJSON(data []byte) error {
+	var value interface{}
+	err := json.Unmarshal(data, &value)
+	if err != nil {
+		return err
+	}
+	switch val := value.(type) {
+	case float64:
+		*a = gateioMilliSecTime(time.UnixMilli(int64(val)))
+	case int64:
+		*a = gateioMilliSecTime(time.UnixMilli(val))
+	case int32:
+		*a = gateioMilliSecTime(time.UnixMilli(int64(val)))
+	case string:
+		if val == "" {
+			return nil
+		}
+		parsedValue, err := strconv.ParseFloat(val, 64)
+		if err != nil {
+			return err
+		}
+		*a = gateioMilliSecTime(time.UnixMilli(int64(parsedValue)))
+	default:
+		return fmt.Errorf("cannot unmarshal %T into gateioMilliSecTime", val)
+	}
+	return nil
+}
+
+// Time represents a time instance.
+func (a *gateioMilliSecTime) Time() time.Time { return time.Time(*a) }
+
+type gateioTime time.Time
+
+// UnmarshalJSON deserializes json, and timestamp information.
+func (a *gateioTime) UnmarshalJSON(data []byte) error {
```

Comment:
If you want you can just have one data type for these times as well.

```go
	var value interface{}
	err := json.Unmarshal(data, &value)
	if err != nil {
		return err
	}

	var standard int64
	switch val := value.(type) {
	case int64:
		standard = val
	case int32:
		standard = int64(val)
	case float64:
		standard = int64(val)
	case string:
		if val == "" {
			return nil
		}
		var parsedValue float64
		parsedValue, err = strconv.ParseFloat(val, 64)
		if err != nil {
			return err
		}
		standard = int64(parsedValue)
	default:
		return fmt.Errorf("cannot unmarshal %T into gateioTime", val)
	}

	if standard > 9999999999 {
		*a = gateioTime(time.UnixMilli(standard))
	} else {
		*a = gateioTime(time.Unix(standard, 0))
	}

	return nil

```

Reply:
I had the idea of changing the method like this. But, I thought you might not accept it.

Since _9999999999_ is going to be _2286-11-20 17:46:39_ It is good to use that.
Thank you!

Reply:
😆